src/pages/workspace/accounting/xero/XeroMapTrackingCategoryConfigurationPage.tsx

```diff
@@ -1,61 +1,74 @@
+import {useRoute} from '@react-navigation/native';
 import React, {useCallback, useMemo} from 'react';
 import ConnectionLayout from '@components/ConnectionLayout';
 import SelectionList from '@components/SelectionList';
 import RadioListItem from '@components/SelectionList/RadioListItem';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as Connections from '@libs/actions/connections';
-import {getTrackingCategory} from '@libs/actions/connections/ConnectToXero';
 import Navigation from '@libs/Navigation/Navigation';
 import type {WithPolicyProps} from '@pages/workspace/withPolicy';
 import withPolicyConnections from '@pages/workspace/withPolicyConnections';
 import CONST from '@src/CONST';
 import type {TranslationPaths} from '@src/languages/types';
 import ROUTES from '@src/ROUTES';
 
-function XeroMapCostCentersToConfigurationPage({policy}: WithPolicyProps) {
+type RouteParams = {
+    categoryId?: string;
+    categoryName?: string;
+};
+
+const TRACKING_CATEGORIES_KEY = 'trackingCategory_';
+
+function XeroMapTrackingCategoryConfigurationPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
+    const route = useRoute();
+    const params = route.params as RouteParams;
     const styles = useThemeStyles();
-
+    const categoryId = params?.categoryId ?? '';
+    const categoryName = decodeURIComponent(params?.categoryName ?? '');
     const policyID = policy?.id ?? '';
+    const {trackingCategories} = policy?.connections?.xero?.data ?? {};
+    const {mappings} = policy?.connections?.xero?.config ?? {};
 
-    const category = getTrackingCategory(policy, CONST.XERO_CONFIG.TRACKING_CATEGORY_FIELDS.COST_CENTERS);
+    const currentTrackingCategory = trackingCategories?.find((category) => category.id === categoryId);
+    const currentTrackingCategoryValue = currentTrackingCategory ? mappings?.[`${TRACKING_CATEGORIES_KEY}${currentTrackingCategory.id}`] ?? '' : '';
 
     const optionsList = useMemo(
         () =>
             Object.values(CONST.XERO_CONFIG.TRACKING_CATEGORY_OPTIONS).map((option) => ({
                 value: option,
                 text: translate(`workspace.xero.trackingCategoriesOptions.${option.toLowerCase()}` as TranslationPaths),
                 keyForList: option,
-                isSelected: option.toLowerCase() === category?.value?.toLowerCase(),
+                isSelected: option === currentTrackingCategoryValue,
             })),
-        [translate, category],
+        [translate, currentTrackingCategoryValue],
     );
 
     const updateMapping = useCallback(
         (option: {value: string}) => {
-            if (option.value !== category?.value) {
+            if (option.value !== categoryName) {
```

Comment:
This change caused a regression https://github.com/Expensify/App/issues/53128

We should prevent calling update xero API when selecting the selected value #53116